Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM rust:1.69-bullseye as rust-build-stage
```

Comment:
is there a reason not to use `latest` here, so we don't have to bump the version in the future?

Reply:
Fair question. Using `:latest` is seen as an anti-pattern for a few reasons. Here are some of them:

1. You're dependent on the build process of the base image distributor keeping that latest tag up to date. You don't really know what version you're getting now or later, or even that it's the latest version actually.
2. Your builds aren't deterministic. The image you build might be different from what I get when I build the next day. 
3. If the docker build starts to fail, but you didn't change your Dockerfile, it's hard to troubleshoot that. How do you find out what the base image version was previously, when the build worked? What is it now, when it's failing?
4. You aren't in control of what you ship to your users. You blindly ship whatever updates happened up stream, whether you need them or not, whether they make sense for you or not.

I definitely sympathize with resistance to adding manual toil updating versions. In my experience, that can be addressed with dependency management (like dependabot, or Renovatebot) in conjunction with build tests in CI/CD. If you are anxious about those ongoing maintenance costs, I could possibly help put together a quick-and-dirty docker build workflow in GitHub Actions for you.

Reply:
Thanks, that definitely makes sense. I was just thinking that *maybe*, those Rust images are relatively well maintained and in principle, updates to the Rust version shouldn't break backwards compatibility, i.e. `vivid` should continue building. But I'm okay with leaving the hardcoded version in. Having a CI check that the image still builds would be great. I don't think there is a need for an automated update mechanism at the moment.

Reply:
I added a basic docker build workflow here: 4d1cfb746f12777116b20c33d17fa07d9906e393

For security reasons, github won't run it here until you merge it. But you can see it running on my fork: https://github.com/rpdelaney/vivid/actions